File: backend/app/models/endpoint.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, DateTime, Text
from app.database.base import Base


class Endpoint(Base):
    __tablename__ = "endpoints"

    id = Column(Integer, primary_key=True, index=True)
    endpoint_id = Column(String(100), unique=True, index=True, nullable=False)
    hostname = Column(String(100), nullable=False, index=True)
    operating_system = Column(String(100), default="Windows 11")
    platform = Column(String(50), default="windows")  # windows, darwin, linux
    ip_address = Column(String(50), nullable=True)
    tailscale_ip = Column(String(50), nullable=True)
    status = Column(String(20), default="Online")  # Online, Offline, Degraded
    last_seen = Column(DateTime, default=datetime.utcnow, index=True)
    collector_version = Column(String(50), default="2.0.0")
    risk_level = Column(String(20), default="Low")  # Low, Medium, High, Critical
    event_count = Column(Integer, default=0)
    incident_count = Column(Integer, default=0)
    registered_at = Column(DateTime, default=datetime.utcnow)

File: frontend/types/endpoint.ts
export interface Endpoint {
  id: number;
  endpoint_id: string;
  hostname: string;
  operating_system: string;
  platform: string;
  ip_address?: string | null;
  tailscale_ip?: string | null;
  status: "Online" | "Offline" | "Degraded" | string;
  last_seen: string;
  collector_version: string;
  risk_level: "Low" | "Medium" | "High" | "Critical" | string;
  event_count: number;
  incident_count: number;
  registered_at: string;
}

export interface EndpointSummary {
  total_endpoints: number;
  online_endpoints: number;
  offline_endpoints: number;
  high_risk_endpoints: number;
  endpoints: Endpoint[];
}
